src/libsync/platform_mac.mm

```diff
@@ -29,53 +37,45 @@ @implementation OwnAppDelegate {
 
 - (BOOL)applicationShouldHandleReopen:(NSApplication *)sender hasVisibleWindows:(BOOL)flag
 {
-    if (auto app = qobject_cast<OCC::Application *>(QApplication::instance()))
-        app->showSettingsDialog();
+    if (auto *app = QApplication::instance()) {
+        QMetaObject::invokeMethod(app, "showSettingsWindow", Qt::QueuedConnection);
+    } else {
+        qDebug() << "Failed to call showSettingsWindow slot";
+    }
     return YES;
 }
 
 @end
 
 namespace OCC {
 
-// implemented in platform_mac_deprecated.mm
-void migrateLaunchOnStartup();
-
-class MacPlatform : public Platform
+class MacPlatformPrivate
 {
 public:
-    MacPlatform();
-    ~MacPlatform() override;
-
-    void migrate() override;
-
-private:
-    QMacAutoReleasePool _autoReleasePool;
-    OwnAppDelegate *_appDelegate;
+    QMacAutoReleasePool autoReleasePool;
+    OwnAppDelegate *appDelegate;
 };
 
 MacPlatform::MacPlatform()
 {
+    Q_D(MacPlatform);
+
     NSApplicationLoad();
-    _appDelegate = [[OwnAppDelegate alloc] init];
-    [[NSApplication sharedApplication] setDelegate:_appDelegate];
+    d->appDelegate = [[OwnAppDelegate alloc] init];
+    [[NSApplication sharedApplication] setDelegate:d->appDelegate];
```

Comment:
This doesn't work, I'm not sure why. @erikjv can you help?

Reply:
It segfaults due to an allocation error. Please try to run either `owncloud` or `owncloudcmd`.

Reply:
I assume it is using the wrong objective c syntax to access the pointer?

Reply:
The `d_ptr` is never initialised. Do something like:
```
MacPlatform::MacPlatform()
    : d_ptr(new MacPlatformPrivate)
{
```

Reply:
Btw the ObjC bits are fine.